c7n/resources/redshiftparamgroup.py

```diff
@@ -0,0 +1,146 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import botocore.exceptions
+
+from c7n.actions import ActionRegistry, BaseAction
+from c7n.filters import FilterRegistry
+from c7n.filters import ValueFilter
+from c7n.manager import resources
+from c7n.query import QueryResourceManager, TypeInfo
+from c7n.utils import (
+    type_schema, local_session)
+from c7n.tags import universal_augment
+
+MAX_RETRY_COUNT = 10
+
+logger = logging.getLogger('custodian.redshift-param-group')
+
+pg_filters = FilterRegistry('redshift-param-group.filters')
+pg_actions = ActionRegistry('redshift-param-group.actions')
+
+
+@resources.register('redshift-param-group')
+class RedshiftParamGroup(QueryResourceManager):
+    """Resource manager for Redshift parameter groups.
+    """
+
+    class resource_type(TypeInfo):
+        service = 'redshift'
+        arn_separator = ":"
+        enum_spec = ('describe_cluster_parameter_groups', 'ParameterGroups', None)
+        name = id = 'ParameterGroupName'
+        filter_name = 'ParameterGroupName'
+        filter_type = 'scalar'
+        date = 'ClusterCreateTime'
+        dimension = 'ParameterGroupName'
+        cfn_type = config_type = "AWS::Redshift::ClusterParameterGroup"
+
+    augment = universal_augment
+
+    filter_registry = pg_filters
+    action_registry = pg_actions
+
+
+@pg_filters.register('check-require-ssl-status')
+class CheckRequireSSLStatusFilter(ValueFilter):
```

Comment:
Resource used "describe_cluster_parameter_groups" which does not provide "require_ssl" configuration, to get this need to call API "describe_cluster_parameters" with "ParameterGroupName".

Reply:
@thisisshi fwiw, this setup does mirror the extant rds and rds clusterparameter group setup afaics. agreed though that it might be worth separately looking at doing a custom augment here to make parameter matching easier.

Reply:
@thisisshi @kapilt do I need to update anything? awaiting for your approval.